azurelinuxagent/common/protocol/hostplugin.py

```diff
@@ -0,0 +1,109 @@
+# Microsoft Azure Linux Agent
+#
+# Copyright 2014 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.4+ and Openssl 1.0+
+#
+
+from azurelinuxagent.common.protocol.wire import *
+from azurelinuxagent.common.utils import textutil
+
+HOST_PLUGIN_PORT = 32526
+URI_FORMAT_GET_API_VERSIONS = "http://{0}:{1}/versions"
+URI_FORMAT_PUT_VM_STATUS = "http://{0}:{1}/status"
+URI_FORMAT_PUT_LOG = "http://{0}:{1}/vmAgentLog"
+API_VERSION = "2015-09-01"
+
+
+class HostPluginProtocol(object):
+    def __init__(self, endpoint):
+        if endpoint is None:
+            raise ProtocolError("Host plugin endpoint not provided")
+        self.is_initialized = False
+        self.is_available = False
+        self.api_versions = None
+        self.endpoint = endpoint
+
+    def initialize(self):
+        self.api_versions = self.get_api_versions()
+        self.is_available = self.api_versions is not None and API_VERSION in self.api_versions
+        self.is_initialized = True
```

Comment:
@hglkrijger Bug then in the getter. Getters should always return legal values, otherwise calling code gets heavily littered with those kinds of checks. Getter should return an empty list in that case.